internal/worker/modelworkermanager/modelworkermanager.go

```diff
@@ -226,8 +226,10 @@ func (m *modelWorkerManager) loop() error {
 }
 
 func (m *modelWorkerManager) modelChanged(modelUUID string) error {
-	model, release, err := m.config.Controller.Model(modelUUID)
-	if errors.Is(err, errors.NotFound) {
+	ctx := context.Background()
```

Comment:
Pass in the context. Get a scoped context on line 219 before the for loop and use it